internal/server/caseload.go

```diff
@@ -38,3 +49,32 @@ func (cv CaseloadVars) buildUrl(team string) string {
 func (cv CaseloadVars) GetTeamUrl(team sirius.Team) string {
 	return cv.buildUrl(team.Selector)
 }
+
+func (cv CaseloadVars) GetReportDueDate(order []sirius.Order) string {
+	return order[0].LatestAnnualReport.DueDate
+}
+
+func (cv CaseloadVars) GetClientStatus(orders []sirius.Order) string {
+	var clientStatus string
+	for _, s := range orders {
+		if s.OrderStatus.Label == "Active" {
+			return "Active"
+		}
+	}
+	for _, s := range orders {
+		if s.OrderStatus.Label == "Open" {
+			return "Open"
+		}
+	}
+	for _, s := range orders {
+		if s.OrderStatus.Label == "Closed" {
+			return "Closed"
+		}
+	}
+	for _, s := range orders {
+		if s.OrderStatus.Label == "Duplicate" {
+			return "Duplicate"
+		}
+	}
+	return clientStatus
```

Comment:
This is quite inefficient, as it iterates over the same array four times. You can rewrite it to avoid that using a map (not tested):
```
func (cv CaseloadVars) GetClientStatus(orders []sirius.Order) string {
	orderStatuses := make(map[string]string)

	for _, s := range orders {
		label := s.OrderStatus.Label
		orderStatuses[label] = label
	}

	statuses := []string{"Active", "Open", "Closed", "Duplicate"}
	for _, status := range statuses {
		if _, found := orderStatuses[status]; found {
			return status
		}
	}

	return ""
}
```

Reply:
⭐